x-pack/plugins/reporting/server/lib/event_logger/logger.ts

```diff
@@ -27,9 +28,39 @@ export interface ExecutionCompleteMetrics {
   byteSize: number;
 }
 
+const getEventLog = (logger: LevelLogger) => ({
+  getLogger(lProperties: Partial<IEvent>): IEventLogger {
+    return (function () {
+      let start: Date | undefined;
+      let end: Date | undefined;
+      let duration: number | undefined;
+
+      return {
+        logEvent(properties: IEvent) {
+          if (start && !end) {
+            end = start ? new Date() : undefined;
+          }
+          duration = end && start ? end.valueOf() - start.valueOf() : undefined;
+          const timing: Partial<IEvent> = {
+            event: { duration, start: start?.toISOString(), end: end?.toISOString() },
```

Comment:
This should include other fields that can be derived, such as Kibana UUID and maybe `@timestamp`

Reply:
Since this is an 8.1 blocker, the changes should be kept minimal. Extra fields can always be added later, if they're really needed.